src/query_engine/query_engine.cpp

```diff
@@ -822,48 +824,45 @@ silo::result_s silo::execute_query(const silo::Database& db, const std::string&
          assert(action["groupByFields"].IsArray());
          std::vector<std::string> groupByFields;
          for (const auto& it : action["groupByFields"].GetArray()) {
-            groupByFields.push_back(it.GetString());
+            groupByFields.emplace_back(it.GetString());
          }
          if (strcmp(action_type, "Aggregated") == 0) {
          } else if (strcmp(action_type, "List") == 0) {
          } else if (strcmp(action_type, "Mutations") == 0) {
          } else {
-            ret.return_message = "Unknown action ";
-            ret.return_message += action_type;
+            query_result.queryResult = response::ErrorResult{"Unknown action", std::string(action_type) + " is not a valid action"};
          }
       } else {
          if (strcmp(action_type, "Aggregated") == 0) {
             unsigned count = execute_count(db, partition_filters);
-            ret.return_message = "{\"count\": " + std::to_string(count) + "}";
+            query_result.queryResult = response::AggregationResult{count};
          } else if (strcmp(action_type, "List") == 0) {
          } else if (strcmp(action_type, "Mutations") == 0) {
             double min_proportion = 0.02;
             if (action.HasMember("minProportion") && action["minProportion"].IsDouble()) {
                if (action["minProportion"].GetDouble() <= 0.0) {
-                  ret.return_message = "{\"message\": \"minProportion must be in interval (0.0,1.0]\"}";
+                  query_result.queryResult = response::ErrorResult{"Invalid proportion", "minProportion must be in interval (0.0,1.0]"};
+                  return query_result;
                }
                min_proportion = action["minProportion"].GetDouble();
             }
             std::vector<mutation_proportion> mutations = execute_mutations(db, partition_filters, min_proportion, perf_out);
-            ret.return_message = "";
-            for (auto& s : mutations) {
-               ret.return_message += "{\"mutation\":\"";
-               ret.return_message += s.mut_from;
-               ret.return_message += std::to_string(s.position);
-               ret.return_message += s.mut_to;
-               ret.return_message += "\",\"proportion\":" + std::to_string(s.proportion) +
-                  ",\"count\":" + std::to_string(s.count) + "},";
-            }
+
+            std::vector<response::MutationProportion> mutationProportions(mutations.size());
+            std::transform(mutations.begin(), mutations.end(), mutationProportions.begin(), [](mutation_proportion mutation_proportion) {
```

Comment:
Could we rename the struct mutation_proportion? (Maybe to MutationProportion) This looks like errors are down to be happening, because it is unclear if the variable or the struct function is called.